### PR TITLE
Use Console as proxy for share object logic

### PR DESCRIPTION
**Pull request by cesnietor**
solves: https://github.com/minio/console/issues/3255

Share File URL now points to Console to facilitate sharing the url in case MinIO endpoint is not exposed.

Changes:
- Update how we return the share url pointing now to Object Browser API instead of MinIO endpoint
- Adds new endpoint to validate share link url (base64encoded) and download file
- unittests to validate url

How it looks:
Screenshot 2024-04-05 at 2 49 32 PM

When url expired:
Screenshot 2024-04-05 at 2 50 58 PM

### Test steps:
With an existing bucket, folder and objects.

- Click share file
- Copy and paste the new url in the browser
- The url should contain same url as Object browser
- Object should be downloaded
- Test injecting wrong urls (base64encoded)
- Should return error or forbidden errors if url doesn't point to internal MinIO server

### CODE REVIEW

**Review by bexsoft — DISMISSED** [review]
LGTM, Tested in local env as well as making a custom MinIO build with this fix.

**Comment by prakashsvmx** [comment]
@cesnietor , Do we still need to limit  the number of hours as in the screenshot?

image

File: api/public_objects.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			http.Error(rw, resp.Status, resp.StatusCode)
+			return
+		}
+
+		urlObj, err := url.Parse(*inputURLDecoded)
+		if err != nil {
+			fmt.Printf("error parsing url '%s': %s", *inputURLDecoded, err)
```
Comment (harshavardhana) [inline api/public_objects.go:75]: Frivolous log please remove.

File: api/public_objects.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		// Add the filename
+		_, objectName := url2BucketAndObject(urlObj)
+		escapedName := url.PathEscape(objectName)
+		rw.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", escapedName))
+
+		_, err = io.Copy(rw, resp.Body)
+		if err != nil {
+			fmt.Println("error copying response body:", err)
```
Comment (harshavardhana) [inline api/public_objects.go:87]: Another frivolous log

File: api/public_objects_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
+package api
```
Comment (harshavardhana) [inline api/public_objects_test.go:1]: Missing copyright

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: api/public_objects_test.go (hunk 3 above)
Comment (cesnietor) [reply api/public_objects_test.go:17]: done

File: api/public_objects.go (hunk 2 above)
Comment (cesnietor) [reply api/public_objects.go]: done

**Review by prakashsvmx — APPROVED** [review]
LGTM :+1:

**Review by bexsoft — APPROVED** [review]
LGTM

**Pull request merged**